src/utils/twoFactor.js

```diff
@@ -0,0 +1,265 @@
+        return await this.request(account, request)
+    }
+
+    async rotateKeys(account, addPublicKey, removePublicKey) {
+        const { accountId } = account
+        const request = {
+            receiver_id: accountId,
+            actions: [
+                addKeyAction(account, addPublicKey, accountId),
+                deleteKeyAction(removePublicKey)
+            ]
+        }
+        return await this.request(account, request)
+    }
+
+    async signAndSendTransactions(account, transactions) {
+        for (let { receiverId, nonce, blockHash, actions } of transactions) {
+            actions = actions.map((a) => {
+                const action = {
+                    ...a[a.enum],
+                    type: actionTypes[a.enum],
+                }
+                if (action.gas) action.gas = action.gas.toString()
+                if (action.deposit) action.deposit = action.deposit.toString()
+                if (action.args) action.args = btoa(new TextDecoder().decode(new Uint8Array(action.args)))
+                if (action.methodName) {
+                    action.method_name = action.methodName
+                    delete action.methodName
+                }
+                return action
+            })
+            await this.request(account, { receiver_id: receiverId, actions })
+        }
+    }
+
+    async sendRequest(accountId, method, requestId = -1, data = {}) {
+        if (!accountId) accountId = this.wallet.accountId
+        if (!method) method = await this.get2faMethod()
+        // add request to local storage
+        setRequest({ accountId, requestId, data })
+        try {
+            await this.wallet.postSignedJson('/2fa/send', {
+                accountId,
+                method,
+                requestId,
+                data
+            })
+        } catch (e) {
+            throw(e)
+        }
+        if (requestId !== -1) {
+            return store.dispatch(promptTwoFactor(true)).payload.store.promise
+        }
+    }
+
+    async deployMultisig() {
+        const accountData = await this.wallet.loadAccount()
+        const contractBytes = new Uint8Array(await (await fetch('/multisig.wasm')).arrayBuffer())
+        const { accountId } = accountData
+        const account = this.wallet.getAccount(accountId)
+        const accountKeys = await account.getAccessKeys();
+        const recoveryMethods = await this.wallet.getRecoveryMethods()
+        const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
+        const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
+        fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+        const getAccessKey = await this.wallet.postSignedJson('/2fa/getAccessKey', {
+            accountId
+        })
+        if (!getAccessKey || !getAccessKey.success || !getAccessKey.publicKey) {
```

Comment:
Is there any reason why not just return error from contract helper so that it doesn't require additional check here?

Reply:
done and cleaning up some of these responses from contract helper